posthog/models/team/team.py

```diff
@@ -197,21 +199,22 @@ def get_effective_membership_level(self, user_id: int) -> Optional["Organization
 
     @property
     def actor_on_events_querying_enabled(self) -> bool:
-        # Two part gate to ensure we don't accidentally enable this for everyone prematurely.
-        # Nor should we start making excessive requests to the API to get results.
-        # Temporary, until we're convinced the library works well enough.
-        enabled_teams = get_list(get_instance_setting("ENABLE_ACTOR_ON_EVENTS_TEAMS"))
-        if len(enabled_teams) > 0:
-            return str(self.pk) in enabled_teams or "all" in enabled_teams
-
-        # only when instance setting list is empty, use the feature flag
-        return posthoganalytics.feature_enabled(
-            "person-on-events-enabled",
-            str(self.uuid),
-            groups={"project": str(self.uuid)},
-            group_properties={"project": {"id": str(self.pk)}},
-            only_evaluate_locally=True,
-        )
+        # on PostHog Cloud, use the feature flag
+        if settings.MULTI_TENANCY:
+            return posthoganalytics.feature_enabled(
+                "person-on-events-enabled",
+                str(self.uuid),
+                groups={"project": str(self.uuid)},
+                group_properties={"project": {"id": str(self.pk)}},
+                only_evaluate_locally=True,
+            )
+
+        # If the async migration is not complete, don't enable actor on events querying.
+        if not person_on_events_ready():
+            return False
+
+        # on self-hosted, use the instance setting
+        return get_instance_setting("PERSON_ON_EVENTS_ENABLED")
```

Comment:
you can now make this shorter
```suggestion
        # on self-hosted, make sure the migration is completed and use the instance setting
        return person_on_events_ready() and get_instance_setting("PERSON_ON_EVENTS_ENABLED")
```